File: 프로그래머스/2/12941. 최솟값 만들기/최솟값 만들기.py
def solution(A,B):
    answer = 0
    asc_A, des_B = sorted(A), sorted(B, reverse=True)
    
    for i in range(len(A)):
        answer += asc_A[0] * des_B[0]
        asc_A.pop(0)
        des_B.pop(0)
        
    return answer

File: 프로그래머스/2/12941. 최솟값 만들기/최솟값 만들기.js
function solution(A,B){
    let answer = 0;
    A.sort((a, b) => a - b);
    B.sort((a, b) => b - a);
    
    A.forEach((v, i) => {
        answer += v * B[i]
    });
    return answer;
}

/*
A는 오름차순 정렬, B는 내림차순 정렬
n번째 위치씩 각각 곱하여 더하기
*/